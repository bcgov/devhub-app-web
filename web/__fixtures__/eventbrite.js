/*
Copyright 2019 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/

// event brite sample data

const EVENT_BRITE_EVENT = {
  internal: {
    content:
      '{"name":{"text":"Lunch & Lab - How to Safely Navigate Open Source Licensing in your Projects","html":"Lunch &amp; Lab - How to Safely Navigate Open Source Licensing in your Projects"},"description":{"text":"WHEN? Friday, April 6, 12-1 pm. CSI Lab Commons, 1012 Douglas. 3rd floor. Come hungry. There will be snacks. Everyone welcome.\\nYou have an existing software application and you want to “crowd source” a fix/update. Maybe you have a new application build that will benefit from leveraging existing open source software, or maybe all you have is an idea in need of a solution. Whether the goal is to make existing government-owned software available under an open source license or bring open source code in, this talk will help you understand what you need to know to comply with government policies for success with your project. We will also be sharing some “lessons learned” and field any questions you may have.\\nBio: Lori Kernaghan is an Intellectual Property Manager with the Intellectual Property Program, Ministry of Citizens’ Services. She has 15 years’ experience as an intellectual property professional. Lori joined the BC Government in 2008 and for the past 10 years has been providing ministry and core agency clients across government with a variety of intellectual property management services, including licensing of intellectual property. Lori has worked on a number of key initiatives in the BC Government, including the Open Information and Open Data Policy, the BC Open Source Development Employee Guide (GitHub) and the development of social media guidelines. \\nRecommended (but optional) Pre-Reading: BC Policy Framework For GitHub\\nWhat is the CSI Lab?\\nThe CSI Lab is a creative space where multi-disciplinary teams of public servants can learn new ways of working using technology as an enabler to build a modern public service.","html":"<P><STRONG>WHEN? Friday, April 6, 12-1 pm. CSI Lab Commons, 1012 Douglas. 3rd floor. Come hungry. There will be snacks. Everyone welcome.</STRONG></P>\\n<P>You have an existing software application and you want to “crowd source” a fix/update. Maybe you have a new application build that will benefit from leveraging existing open source software, or maybe all you have is an idea in need of a solution. Whether the goal is to make existing government-owned software available under an open source license or bring open source code in, this talk will help you understand what you need to know to comply with government policies for success with your project. We will also be sharing some “lessons learned” and field any questions you may have.</P>\\n<P><SPAN><STRONG>Bio</STRONG>:</SPAN> Lori Kernaghan is an Intellectual Property Manager with the Intellectual Property Program, Ministry of Citizens’ Services. She has 15 years’ experience as an intellectual property professional. Lori joined the BC Government in 2008 and for the past 10 years has been providing ministry and core agency clients across government with a variety of intellectual property management services, including licensing of intellectual property. Lori has worked on a number of key initiatives in the BC Government, including the Open Information and Open Data Policy, the BC Open Source Development Employee Guide (GitHub) and the development of social media guidelines. </P>\\n<P><SPAN><STRONG>Recommended (but optional) Pre-Reading:</STRONG> </SPAN><A HREF=\\"https://github.com/bcgov/BC-Policy-Framework-For-GitHub\\" TARGET=\\"_blank\\" REL=\\"nofollow noopener noreferrer\\">BC Policy Framework For GitHub</A></P>\\n<P><STRONG>What is the CSI Lab?</STRONG></P>\\n<P>The CSI Lab is a creative space where multi-disciplinary teams of public servants can learn new ways of working using technology as an enabler to build a modern public service.</P>"},"id":"44648171884","url":"https://www.eventbrite.ca/e/lunch-lab-how-to-safely-navigate-open-source-licensing-in-your-projects-tickets-44648171884","start":{"timezone":"America/Vancouver","local":"2018-04-06T12:00:00","utc":"2018-04-06T19:00:00Z"},"end":{"timezone":"America/Vancouver","local":"2018-04-06T13:00:00","utc":"2018-04-06T20:00:00Z"},"organization_id":"228490647317","created":"2018-03-28T18:27:40Z","changed":"2018-04-07T03:48:10Z","published":"2018-03-28T18:29:56Z","capacity":50,"capacity_is_custom":false,"status":"completed","currency":"CAD","listed":false,"shareable":false,"invite_only":false,"online_event":false,"show_remaining":false,"tx_time_limit":480,"hide_start_date":false,"hide_end_date":false,"locale":"en_CA","is_locked":false,"privacy_setting":"unlocked","is_series":false,"is_series_parent":false,"inventory_type":"limited","is_reserved_seating":false,"show_pick_a_seat":false,"show_seatmap_thumbnail":false,"show_colors_in_seatmap_thumbnail":false,"source":"create_2.0","is_free":true,"version":"3.0.0","summary":"WHEN? Friday, April 6, 12-1 pm. CSI Lab Commons, 1012 Douglas. 3rd floor. Come hungry. There will be snacks. Everyone welcome.\\nYou have an e","logo_id":"42845796","organizer_id":"17134258064","venue_id":"23970647","category_id":null,"subcategory_id":null,"format_id":null,"resource_uri":"https://www.eventbriteapi.com/v3/events/44648171884/","is_externally_ticketed":false,"logo":{"crop_mask":{"top_left":{"x":0,"y":0},"width":794,"height":397},"original":{"url":"https://img.evbuc.com/https%3A%2F%2Fcdn.evbuc.com%2Fimages%2F42845796%2F228490647317%2F1%2Foriginal.jpg?auto=compress&s=a2829f497244deb7ebb5fe91f5c9400d","width":794,"height":1123},"id":"42845796","url":"https://img.evbuc.com/https%3A%2F%2Fcdn.evbuc.com%2Fimages%2F42845796%2F228490647317%2F1%2Foriginal.jpg?h=200&w=450&auto=compress&rect=0%2C0%2C794%2C397&s=ae5a86a02c4ef24f4be03bdda60d27a1","aspect_ratio":"2","edge_color":null,"edge_color_set":true}}',
    description: 'DEMO EVENT BRITE',
    mediaType: 'text/html',
    type: 'EventbriteEvents',
  },
  name: {
    text: 'DEMO EVENT BRITE DATA - DEVELOPMENT ONLY',
    html: 'DEMO EVENT BRITE DATA - DEVELOPMENT ONLY',
  },
  description: {
    text: 'DEMO EVENT BRITE DATA - DEVELOPMENT ONLY',
    html: 'DEMO EVENT BRITE DATA - DEVELOPMENT ONLY',
  },
  url:
    'https://www.eventbrite.ca/e/lunch-lab-how-to-safely-navigate-open-source-licensing-in-your-projects-tickets-44648171884',
  start: {
    timezone: 'America/Vancouver',
    local: '2018-04-06T12:00:00',
    utc: '2018-04-06T19:00:00Z',
  },
  end: {
    timezone: 'America/Vancouver',
    local: '2018-04-06T13:00:00',
    utc: '2018-04-06T20:00:00Z',
  },
  organizer_id: '17134258064',
  organization_id: '228490647317',
  created: '2018-03-28T18:27:40Z',
  changed: '2018-04-07T03:48:10Z',
  published: '2018-03-28T18:29:56Z',
  privacy_setting: 'unlocked',
  capacity: 50,
  category_id: null,
  currency: 'CAD',
  capacity_is_custom: false,
  locale: 'en_CA',
  shareable: true,
  logo: {
    id: 'DEMO',
    url:
      'https://img.evbuc.com/https%3A%2F%2Fcdn.evbuc.com%2Fimages%2F42845796%2F228490647317%2F1%2Foriginal.jpg?h=200&w=450&auto=compress&rect=0%2C0%2C794%2C397&s=ae5a86a02c4ef24f4be03bdda60d27a1',
    crop_mask: {
      width: 794,
      height: 397,
      top_left: {
        x: 0,
        y: 0,
      },
    },
    original: {
      url:
        'https://img.evbuc.com/https%3A%2F%2Fcdn.evbuc.com%2Fimages%2F42845796%2F228490647317%2F1%2Foriginal.jpg?auto=compress&s=a2829f497244deb7ebb5fe91f5c9400d',
      width: 794,
      height: 1123,
      localFile: {
        id: 'e87c2530-f76c-5e28-bf50-f4e483371cd0',
      },
    },
    aspect_ratio: '2',
    edge_color: null,
    edge_color_set: true,
  },
  logo_id: 'DEMO@',
  organization: 'CSI Lab',
  venue_id: 'DEMO_VENUE',
  format_id: null,
  resource_uri: 'https://www.eventbriteapi.com/v3/events/DEMO/',
  is_externally_ticketed: false,
  series_id: null,
  venue: {
    id: '23970647',
    address: {
      address_1: '3rd Floor, 1012 Douglas Street',
      address_2: null,
      city: 'Victoria',
      region: 'BC',
      postal_code: 'V8W 2C3',
      country: 'CA',
      latitude: '48.4242409',
      longitude: '-123.36581739999997',
      localized_address_display: '3rd Floor, 1012 Douglas Street, Victoria, BC V8W 2C3',
      localized_area_display: 'Victoria, BC',
    },
    resource_uri: 'https://www.eventbriteapi.com/v3/venues/23970647/',
    name: '1012 Douglas St',
    latitude: '48.4242409',
    longitude: '-123.36581739999997',
    internal: {
      type: 'EventbriteVenues',
      content:
        '{"address":{"address_1":"3rd Floor, 1012 Douglas Street","address_2":null,"city":"Victoria","region":"BC","postal_code":"V8W 2C3","country":"CA","latitude":"48.4242409","longitude":"-123.36581739999997","localized_address_display":"3rd Floor, 1012 Douglas Street, Victoria, BC V8W 2C3","localized_area_display":"Victoria, BC","localized_multi_line_address_display":["3rd Floor, 1012 Douglas Street","Victoria, BC V8W 2C3"]},"resource_uri":"https://www.eventbriteapi.com/v3/venues/23970647/","id":"23970647","age_restriction":null,"capacity":null,"name":"1012 Douglas St","latitude":"48.4242409","longitude":"-123.36581739999997"}',
      contentDigest: 'b72d19dba8d990adf5455d65ff6adae8',
      owner: 'gatsby-source-eventbrite',
    },
  },
};

module.exports = {
  EVENT_BRITE_EVENT,
};
