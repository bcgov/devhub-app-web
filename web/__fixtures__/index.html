<!-- DO NOT DELETE -->
<!-- This file is used for any future integration tests between the devhub and the react-rehype/component preview component -->
<!DOCTYPE html>
<html lang="en">
<head>
    <link href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.1.3/css/bootstrap-reboot.min.css" rel="stylesheet">
    <link href="https://fonts.googleapis.com/css?family=Noto+Sans" rel="stylesheet">
    <style>
      header {
        background-color: blue;
        color: white;
        padding: 35px;
        border-bottom: 1px solid yellow;
      }
    </style>
    <title>Sample Header</title>
</head>

<!--
  All in-line CSS is specific to this sample; it can and should be ignored.
 -->

<body>
  <header>
    <div class="banner">
        <a href="https://gov.bc.ca" alt="British Columbia">
          <img src="../assets/images/logo-banner.svg" alt="Go to the Government of British Columbia website" />
        </a>
        <h1>Hello British Columbia</h1>
    </div>
    <div class="other">
    <!-- 
      This place is for anything that needs to be right aligned
      beside the logo.  
    -->
      &nbsp;
    </div>
    </div>
  </header>
</body>
</html>
