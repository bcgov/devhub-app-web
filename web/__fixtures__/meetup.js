/*
Copyright 2019 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Derek Siemens
*/

// Meetup sample data

const MEETUP_GROUP = {
  internal: {
    content:
      '{"id":28512032,"name":"BC DevOps Commons","status":"active","link":"https://www.meetup.com/DevOps-Commons/","urlname":"DevOps-Commons","description":"<p>We are reviving the DevOps Commons!&nbsp;<span>This recurring Meetup is a cross government community group made up of government employees, broader public sector employees and contractors.&nbsp; We are committed to excellence in the continuous improvement of development teams, delivery processes and technology solutions. We are looking forward to learning and collaborating on all things DevOps with you.</span></p>","created":1526511874000,"city":"Victoria","untranslated_city":"Victoria","country":"CA","localized_country_name":"Canada","localized_location":"Victoria, BC","state":"BC","join_mode":"open","visibility":"public_limited","lat":48.43,"lon":-123.36,"members":211,"organizer":{"id":213475102,"name":"Luke Gonis","bio":""},"who":"Members","group_photo":{"id":471194841,"highres_link":"https://secure.meetupstatic.com/photos/event/3/9/f/9/highres_471194841.jpeg","photo_link":"https://secure.meetupstatic.com/photos/event/3/9/f/9/600_471194841.jpeg","thumb_link":"https://secure.meetupstatic.com/photos/event/3/9/f/9/thumb_471194841.jpeg","type":"event","base_url":"https://secure.meetupstatic.com"},"key_photo":{"id":471194847,"highres_link":"https://secure.meetupstatic.com/photos/event/3/9/f/f/highres_471194847.jpeg","photo_link":"https://secure.meetupstatic.com/photos/event/3/9/f/f/600_471194847.jpeg","thumb_link":"https://secure.meetupstatic.com/photos/event/3/9/f/f/thumb_471194847.jpeg","type":"event","base_url":"https://secure.meetupstatic.com"},"timezone":"Canada/Pacific","category":{"id":34,"name":"Tech","shortname":"tech","sort_name":"Tech"},"meta_category":{"id":292,"shortname":"tech","name":"Tech","sort_name":"Tech","photo":{"id":450131949,"highres_link":"https://secure.meetupstatic.com/photos/event/2/e/a/d/highres_450131949.jpeg","photo_link":"https://secure.meetupstatic.com/photos/event/2/e/a/d/600_450131949.jpeg","thumb_link":"https://secure.meetupstatic.com/photos/event/2/e/a/d/thumb_450131949.jpeg","type":"event","base_url":"https://secure.meetupstatic.com"},"category_ids":[34]}}',
    contentDigest: 'ac57dc5ab427bcf398a162753f4d0aaf',
    description: 'DEMO MEETUP DATA - DEVELOPER ONLY ',
    type: 'MeetupGroup',
  },
  name: 'BC DevOps Commons',
  status: 'active',
  link: 'https://www.meetup.com/DevOps-Commons/',
  urlname: 'DevOps-Commons',
  description: 'DEMO MEETUP DATA - DEVELOPER ONLY',
  created: 1526511874000,
  city: 'Victoria',
  untranslated_city: 'Victoria',
  country: 'CA',
  localized_country_name: 'Canada',
  localized_location: 'Victoria, BC',
  state: 'BC',
  join_mode: 'open',
  visibility: 'public_limited',
  lat: 48.43,
  lon: -123.36,
  members: 211,
  organizer: {
    id: 213475102,
    name: 'Luke Gonis',
    bio: '',
    photo: null,
  },
  who: 'Members',
  group_photo: {
    id: 471194841,
    highres_link: 'https://secure.meetupstatic.com/photos/event/3/9/f/9/highres_471194841.jpeg',
    photo_link: 'https://secure.meetupstatic.com/photos/event/3/9/f/9/600_471194841.jpeg',
    thumb_link: 'https://secure.meetupstatic.com/photos/event/3/9/f/9/thumb_471194841.jpeg',
    type: 'event',
    base_url: 'https://secure.meetupstatic.com',
  },
  key_photo: {
    id: 471194847,
    highres_link: 'https://secure.meetupstatic.com/photos/event/3/9/f/f/highres_471194847.jpeg',
    photo_link: 'https://secure.meetupstatic.com/photos/event/3/9/f/f/600_471194847.jpeg',
    thumb_link: 'https://secure.meetupstatic.com/photos/event/3/9/f/f/thumb_471194847.jpeg',
    type: 'event',
    base_url: 'https://secure.meetupstatic.com',
  },
  timezone: 'Canada/Pacific',
  category: {
    id: 34,
    name: 'Tech',
    shortname: 'tech',
    sort_name: 'Tech',
  },
  meta_category: {
    id: 292,
    shortname: 'tech',
    sort_name: 'Tech',
    category_ids: [34],
    photo: {
      id: 450131949,
      highres_link: 'https://secure.meetupstatic.com/photos/event/2/e/a/d/highres_450131949.jpeg',
      photo_link: 'https://secure.meetupstatic.com/photos/event/2/e/a/d/600_450131949.jpeg',
      thumb_link: 'https://secure.meetupstatic.com/photos/event/2/e/a/d/thumb_450131949.jpeg',
      type: 'event',
      base_url: 'https://secure.meetupstatic.com',
    },
  },
  pro_network: null,
  next_event: null,
  childrenMeetupEvent: [
    {
      id: 'a5293451-4f1d-5518-8099-e2f2bcb25891',
      internal: {
        content:
          '{"created":1561565461000,"duration":3600000,"id":"262643690","name":"Anything but Jenkins, long live Jenkins ","rsvp_limit":41,"date_in_series_pattern":false,"status":"past","time":1562097600000,"local_date":"2019-07-02","local_time":"13:00","updated":1562107379000,"utc_offset":-25200000,"waitlist_count":0,"yes_rsvp_count":31,"venue":{"id":25859211,"name":"BC Government Continuous Service Improvement Lab - Grand Central Station (3rd Floor Common Area)","lat":48.42424011230469,"lon":-123.36581420898438,"repinned":true,"address_1":"3rd Floor 1012 Douglas St.","city":"Victoria","country":"ca","localized_country_name":"Canada","zip":"V8W 2C3","state":"bc"},"group":{"created":1526511874000,"name":"BC DevOps Commons","id":28512032,"join_mode":"open","lat":48.43000030517578,"lon":-123.36000061035156,"urlname":"DevOps-Commons","who":"Members","localized_location":"Victoria, BC","state":"BC","country":"ca","region":"en_US","timezone":"Canada/Pacific"},"link":"https://www.meetup.com/DevOps-Commons/events/262643690/","description":"<p>We love Jenkins, but thereâ€™s got to be a better way... join pipeline experts as we demo and explore Continuous Integration and Continuous Delivery tools and processes.</p> <p>Click here to join remotely: <a href=\\"https://meet.google.com/cmd-fxui-myk\\" class=\\"linkified\\">https://meet.google.com/cmd-fxui-myk</a></p> ","visibility":"public_limited"}',
        contentDigest: 'bc323e4419c8a3eff7a41dea2cfaa4b2',
        description: 'DEMO MEETUP DATA - DEVELOPER ONLY ',
        owner: 'gatsby-source-meetup',
        type: 'MeetupEvent',
      },
      siphon: {
        unfurl: {
          title: 'DEMO MEETUP DATA - DEVELOPER ONLY',
          image: 'meetup',
          description: 'DEMO MEETUP DATA - DEVELOPER ONLY',
        },
        resource: {
          type: 'Events',
          path: 'https://www.meetup.com/DevOps-Commons/events/262643690/',
        },
        id: 'a5293451-4f1d-5518-8099-e2f2bcb25891',
      },
      created: 1561565461000,
      duration: 3600000,
      name: 'Anything but Jenkins, long live Jenkins ',
      rsvp_limit: 41,
      date_in_series_pattern: false,
      status: 'past',
      time: 1562097600000,
      local_date: '2019-07-02',
      local_time: '13:00',
      updated: 1562107379000,
      utc_offset: -25200000,
      waitlist_count: 0,
      yes_rsvp_count: 31,
      venue: {
        id: 25859211,
        name:
          'BC Government Continuous Service Improvement Lab - Grand Central Station (3rd Floor Common Area)',
        lat: 48.42424011230469,
        lon: -123.36581420898438,
        repinned: true,
        address_1: '3rd Floor 1012 Douglas St.',
        city: 'Victoria',
        country: 'ca',
        localized_country_name: 'Canada',
        zip: 'V8W 2C3',
        state: 'bc',
        address_2: null,
        phone: null,
      },
      group: {
        created: 1526511874000,
        name: 'BC DevOps Commons',
        id: 28512032,
        join_mode: 'open',
        lat: 48.43000030517578,
        lon: -123.36000061035156,
        urlname: 'DevOps-Commons',
        who: 'Members',
        localized_location: 'Victoria, BC',
        state: 'BC',
        country: 'ca',
        region: 'en_US',
        timezone: 'Canada/Pacific',
      },
      link: 'https://www.meetup.com/DevOps-Commons/events/262643690/',
      description: 'DEMO MEETUP',
      visibility: 'public_limited',
      fields: {
        name: 'DEMO MEETUP DATA - DEVELOPER ONLY',
        description: 'DEMO MEETUP DATA - DEVELOPER ONLY',
        link: 'https://www.meetup.com/DevOps-Commons/events/262643690/',
        location: '3rd Floor 1012 Douglas St.',
      },
      how_to_find_us: null,
      pro_is_email_shared: null,
      rsvp_close_offset: null,
      fee: null,
      manual_attendance_count: null,
    },
  ],
};

module.exports = {
  MEETUP_GROUP,
};
