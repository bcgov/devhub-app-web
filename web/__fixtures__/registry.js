export const JOURNEY_REGISTRY = [
  {
    name: 'New Developer',
    sourceProperties: {
      stops: [
        {
          stops: [{ id: 10 }],
          id: 11,
        },
        {
          stops: [{ id: 13 }, { id: 14 }],
          id: 12,
        },
      ],
    },
  },
  {
    name: 'New Developer',
    sourceProperties: {
      stops: [
        {
          stops: [{ id: 2 }],
          id: 1,
        },
        {
          stops: [{ id: 4 }, { id: 5 }],
          id: 3,
        },
      ],
    },
  },
];

export const JOURNEY_REGISTRY_MAPPED_AS_TOPIC = [
  {
    name: 'New Developer',
    sourceProperties: {
      sources: [
        { id: 11 },
        {
          id: 10,
          connectsWith: {
            id: 11,
          },
        },
        {
          id: 12,
        },
        {
          id: 13,
          connectsWith: {
            id: 12,
          },
        },
        {
          id: 14,
          connectsWith: {
            id: 12,
          },
        },
      ],
    },
  },
  {
    name: 'New Developer',
    sourceProperties: {
      sources: [
        {
          id: 1,
        },
        {
          id: 2,
          connectsWith: {
            id: 1,
          },
        },
        {
          id: 3,
        },
        {
          id: 4,
          connectsWith: {
            id: 3,
          },
        },
        {
          id: 5,
          connectsWith: {
            id: 3,
          },
        },
      ],
    },
  },
];
