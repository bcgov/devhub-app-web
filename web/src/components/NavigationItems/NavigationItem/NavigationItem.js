/*
Copyright 2018 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/
import React from 'react';
import Link from '../../UI/Link/Link';
import PropTypes from 'prop-types';
import { css } from '@emotion/core';
const NavigationItem = ({ to, text }) => (
  <li>
    <Link
      to={to}
      css={css`
        color: #fff;
        font-size: 0.813em;
        text-transform: capitalize;
        font-weight: normal;
        text-decoration: none;
        :hover {
          color: #fff;
        }
      `}
    >
      {text}
    </Link>
  </li>
);

NavigationItem.propTypes = {
  to: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
};

export default NavigationItem;
