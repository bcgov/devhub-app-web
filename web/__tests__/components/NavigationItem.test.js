/*
Copyright 2018 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/
import React from 'react';
import { shallow } from 'enzyme';
import NavigationItem from '../../src/components/NavigationItems/NavigationItem/NavigationItem';
import { A_NAV_ITEM } from '../../__fixtures__/ui-fixtures';

describe('Primary Footer Component', () => {
  test('it matches snapshot', () => {
    const wrapper = shallow(<NavigationItem to={A_NAV_ITEM.to} text={A_NAV_ITEM.text} />);
    expect(wrapper).toMatchSnapshot();
  });
});
