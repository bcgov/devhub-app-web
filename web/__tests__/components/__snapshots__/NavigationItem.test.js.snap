// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primary Footer Component it matches snapshot 1`] = `
<li>
  <GatsbyLink
    activeClassName=""
    activeStyle={Object {}}
    css={
      Object {
        "map": "/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIk5hdmlnYXRpb25JdGVtLmpzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQXlCYyIsImZpbGUiOiJOYXZpZ2F0aW9uSXRlbS5qcyIsInNvdXJjZXNDb250ZW50IjpbIi8qXG5Db3B5cmlnaHQgMjAxOCBQcm92aW5jZSBvZiBCcml0aXNoIENvbHVtYmlhXG5cbkxpY2Vuc2VkIHVuZGVyIHRoZSBBcGFjaGUgTGljZW5zZSwgVmVyc2lvbiAyLjAgKHRoZSBcIkxpY2Vuc2VcIik7XG55b3UgbWF5IG5vdCB1c2UgdGhpcyBmaWxlIGV4Y2VwdCBpbiBjb21wbGlhbmNlIHdpdGggdGhlIExpY2Vuc2UuXG5Zb3UgbWF5IG9idGFpbiBhIGNvcHkgb2YgdGhlIExpY2Vuc2UgYXQgXG5cbiAgIGh0dHA6Ly93d3cuYXBhY2hlLm9yZy9saWNlbnNlcy9MSUNFTlNFLTIuMFxuXG5Vbmxlc3MgcmVxdWlyZWQgYnkgYXBwbGljYWJsZSBsYXcgb3IgYWdyZWVkIHRvIGluIHdyaXRpbmcsIHNvZnR3YXJlXG5kaXN0cmlidXRlZCB1bmRlciB0aGUgTGljZW5zZSBpcyBkaXN0cmlidXRlZCBvbiBhbiBcIkFTIElTXCIgQkFTSVMsXG5XSVRIT1VUIFdBUlJBTlRJRVMgT1IgQ09ORElUSU9OUyBPRiBBTlkgS0lORCwgZWl0aGVyIGV4cHJlc3Mgb3IgaW1wbGllZC5cblNlZSB0aGUgTGljZW5zZSBmb3IgdGhlIHNwZWNpZmljIGxhbmd1YWdlIGdvdmVybmluZyBwZXJtaXNzaW9ucyBhbmRcbmxpbWl0YXRpb25zIHVuZGVyIHRoZSBMaWNlbnNlLlxuXG5DcmVhdGVkIGJ5IFBhdHJpY2sgU2ltb25pYW5cbiovXG5pbXBvcnQgUmVhY3QgZnJvbSAncmVhY3QnO1xuaW1wb3J0IExpbmsgZnJvbSAnLi4vLi4vVUkvTGluay9MaW5rJztcbmltcG9ydCBQcm9wVHlwZXMgZnJvbSAncHJvcC10eXBlcyc7XG5pbXBvcnQgeyBjc3MgfSBmcm9tICdAZW1vdGlvbi9jb3JlJztcbmNvbnN0IE5hdmlnYXRpb25JdGVtID0gKHsgdG8sIHRleHQgfSkgPT4gKFxuICA8bGk+XG4gICAgPExpbmtcbiAgICAgIHRvPXt0b31cbiAgICAgIGNzcz17Y3NzYFxuICAgICAgICBjb2xvcjogI2ZmZjtcbiAgICAgICAgZm9udC1zaXplOiAwLjgxM2VtO1xuICAgICAgICB0ZXh0LXRyYW5zZm9ybTogY2FwaXRhbGl6ZTtcbiAgICAgICAgZm9udC13ZWlnaHQ6IG5vcm1hbDtcbiAgICAgICAgdGV4dC1kZWNvcmF0aW9uOiBub25lO1xuICAgICAgICA6aG92ZXIge1xuICAgICAgICAgIGNvbG9yOiAjZmZmO1xuICAgICAgICB9XG4gICAgICBgfVxuICAgID5cbiAgICAgIHt0ZXh0fVxuICAgIDwvTGluaz5cbiAgPC9saT5cbik7XG5cbk5hdmlnYXRpb25JdGVtLnByb3BUeXBlcyA9IHtcbiAgdG86IFByb3BUeXBlcy5zdHJpbmcuaXNSZXF1aXJlZCxcbiAgdGV4dDogUHJvcFR5cGVzLnN0cmluZy5pc1JlcXVpcmVkLFxufTtcblxuZXhwb3J0IGRlZmF1bHQgTmF2aWdhdGlvbkl0ZW07XG4iXX0= */",
        "name": "18jy1yo-NavigationItem",
        "styles": "color:#fff;font-size:0.813em;text-transform:capitalize;font-weight:normal;text-decoration:none;:hover{color:#fff;}label:NavigationItem;",
      }
    }
    to="/"
  >
    banana
  </GatsbyLink>
</li>
`;
